### DISCUSSION TITLE
How would i write a middleware helper?

### DISCUSSION THREAD

**Discussion by ndrswlkr** [mojolicious/mojo.js#93] [Q&A] [answered]
I saw it mentoined somewhere that we could write a middleware helper (for expressjs middleware).
I would like to serve my sveltekit-app from my mojo server.

in an express app that would be as simple as:
```

import { handler } from '../build/handler.js'  // this handler is created by sveltekit at build
app.use( handler )
```

in mojojs i tried:
```

app.get('/*', async ctx => {
   const  req = await ctx.req
   const res = await ctx.res
   const next = ()=>{}
   handler( req, res, next )
})
```

it seems that the request object is fine but the response object may cause problems. i get this errormessage:
```
    res.writeHead(response.status, headers);
	    ^
    TypeError: res.writeHead is not a function
```

i checked the mojo response object and there is no such thing as writeHead.

how can i get the right response object?

thank you

**Comment by kraih** [comment, marked as answer]
Express middleware support is still on the todo list i'm afraid. https://github.com/mojolicious/mojo.js/issues/66

**Reply by ndrswlkr** [reply]
Ok, i see...
so no quick and dirty solution for what i want?

**Reply by kraih** [reply]
Not yet, but i'm adding a way in the next release. Not quite sure about the naming though. https://github.com/mojolicious/mojo.js/commit/96092f4f346beec42a8c3e8b38a3916ad6a4b2ee

**Reply by ndrswlkr** [reply]
cool
